helix-core/src/main/java/org/apache/helix/manager/zk/client/HelixZkClient.java

```diff
@@ -6,12 +6,12 @@
 import org.I0Itec.zkclient.DataUpdater;
 import org.I0Itec.zkclient.IZkChildListener;
 import org.I0Itec.zkclient.IZkDataListener;
-import org.I0Itec.zkclient.IZkStateListener;
 import org.I0Itec.zkclient.serialize.SerializableSerializer;
 import org.I0Itec.zkclient.serialize.ZkSerializer;
 import org.apache.helix.manager.zk.BasicZkSerializer;
 import org.apache.helix.manager.zk.PathBasedZkSerializer;
 import org.apache.helix.manager.zk.ZkAsyncCallbacks;
+import org.apache.helix.manager.zk.zookeeper.IZkStateListener;
```

Comment:
As we discussed, let's do the following default method, so no one need to change code. And we are backward compatible.

  default void subscribeStateChanges(final IZkStateListener listener) {

    subscribeStateChanges(new IZkStateListener() {
      @Override
      public void handleStateChanged(Watcher.Event.KeeperState state)
          throws Exception {

      }

      @Override
      public void handleNewSession(Session)
          throws Exception {
        listener.handleNewSession();
      }

      @Override
      public void handleSessionEstablishmentError(Throwable error)
          throws Exception {

      }
    }
  }

Reply:
This is very cool! I like this as it has better backwards compatibility. Thanks!